lib/charms/mysql/v0/mysql.py

```diff
@@ -321,68 +322,53 @@ def configure_mysqlrouter_user(self, username: str, password: str) -> None:
             raise MySQLConfigureRouterUserError(e.message)
 
     def create_application_database_and_scoped_user(
-        self, database_name: str, username: str, password: str
+        self, database_name: str, username: str, password: str, hostname: str, unit_name: str
     ) -> None:
         """Create an application database and a user scoped to the created database.
 
         Args:
             database_name: The name of the database to create
             username: The username of the scoped user
             password: The password of the scoped user
+            hostname: The hostname of the scoped user
+            unit_name: The name of the unit from which the user will be accessed
 
         Raises MySQLCreateApplicationDatabaseAndScopedUserError
             if there is an issue creating the application database or a user scoped to the database
         """
         create_database_commands = (f"CREATE DATABASE IF NOT EXISTS {database_name}",)
+        user_attributes = {"unit_name": unit_name}
         create_scoped_user_commands = (
-            f"CREATE USER '{username}'@'%' IDENTIFIED BY '{password}'",
-            f"GRANT USAGE ON *.* TO '{username}'@`%`",
-            f"GRANT ALL PRIVILEGES ON `{database_name}`.* TO `{username}`@`%`",
+            f"CREATE USER '{username}'@'{hostname}' IDENTIFIED BY '{password}' ATTRIBUTE '{json.dumps(user_attributes)}'",
+            f"GRANT USAGE ON *.* TO '{username}'@`{hostname}`",
+            f"GRANT ALL PRIVILEGES ON `{database_name}`.* TO `{username}`@`{hostname}`",
         )
 
         try:
             self._run_mysqlcli_script("; ".join(create_database_commands))
             self._run_mysqlcli_script("; ".join(create_scoped_user_commands))
         except MySQLClientError as e:
             logger.exception(
-                f"Failed to create application database {database_name} and scoped user {username}@%"
+                f"Failed to create application database {database_name} and scoped user {username}@{hostname}",
+                exc_info=e,
             )
             raise MySQLCreateApplicationDatabaseAndScopedUserError(e.message)
 
-    def remove_user(self, username: str, hostname: str = "%") -> None:
-        """Remove a mysql user.
-
-        Args:
-            username: The username for the mysql user
-            hostname (optional): The hostname for the mysql user (defaults to %)
-
-        Raises MySQLRemoveUserError
-            if there is an issue removing the mysql user
-        """
-        remove_user_commands = (f"DROP USER IF EXISTS '{username}'@'{hostname}'",)
+    def delete_users_for_unit(self, unit_name: str) -> None:
+        """Delete users for a unit."""
```

Comment:
add `Args:` in docstring

Reply:
Addressed in [b4f569d](https://github.com/canonical/mysql-operator/pull/15/commits/b4f569df9c33bec957c9f3eafd520c9950054b00)